bake/bake_test.go

```diff
@@ -585,3 +585,69 @@ services:
 	require.Equal(t, "./Dockerfile", *m["addon"].Dockerfile)
 	require.Equal(t, "./aws.Dockerfile", *m["aws"].Dockerfile)
 }
+
+func TestReadTargetsSameGroupTarget(t *testing.T) {
+	t.Parallel()
+	ctx := context.TODO()
+
+	f := File{
+		Name: "docker-bake.hcl",
+		Data: []byte(`
```

Comment:
I find JSON a bit nicer for these kind of tests, as you can indent each line as much as needed ;)

However, this test suite in particular has lots of strings with YAML, JSON & HCL... maybe a cleanup is due to move these assets to separate files.